docs/design/2021-03-09-security-enhanced-mode.md

```diff
@@ -0,0 +1,191 @@
+# Proposal:
+
+- Author(s):     [morgo](https://github.com/morgo)
+- Last updated:  March 9, 2021
+- Discussion at: N/A
+
+## Abstract
+
+This document was created to discuss the design of Security Enhanced Mode. It comes from the requirement of [Permission isolation from DBaaS users and DBaaS admin](https://docs.google.com/document/d/1S7-8KqR98DYaE-2NCrPiDWlmUvsHvIZytCE8Ofmbgyg/edit) (internal document).
+
+## Terminology
+
+* **Configuration Option**: The name of a variable as set in a configuration file.
+* **System Variable** (aka sysvar): The name of a variable that is set in a running TiDB server using the MySQL protocol.
+* **Super**: The primary MySQL "admin" privilege, which is intended to be superseded by MySQL’s "dynamic" (fine-grained) privileges starting from MySQL 8.0.
+
+## Background
+
+Currently the MySQL `SUPER` privilege encapsulates a very large set of system capabilities. It does not follow the best practices of allocating _fine grained access_ to users based only on their system-access requirements.
+
+This is particularly problematic in a DBaaS scenario such as TiDB Cloud where the `SUPER` privilege has elements that are required by both end users (TiDB Cloud Customers) and system operations (PingCAP SREs).
+
+The design of Security Enhanced Mode (SEM) takes the approach of:
+
+1. Restricting `SUPER` to a set of capabilities that are safe for end users.
+2. Implementation of dynamic privileges ([issue #22439](https://github.com/pingcap/tidb/issues/22439)).
+
+This approach was requested by product management based on the broad "in the wild" association of `SUPER` as "the MySQL admin privilege". Thus, proposals to create a new lesser-`SUPER` privilege have already been discussed and rejected.
+
+The design and name of "Security Enhanced" is inspired by prior art with SELinux and AppArmor.
+
+## Proposal
+
+A boolean option called `EnableEnhancedSecurity` (default `FALSE`) will be added as a TiDB configuration option.  The following subheadings describe the behavior when `EnableEnhancedSecurity` is set to `TRUE`.
+
+### System Variables
+
+The following system variables will be hidden:
+
+* variable.TiDBDDLSlowOprThreshold,
+* variable.TiDBAllowRemoveAutoInc,
+* variable.TiDBCheckMb4ValueInUTF8,
+* variable.TiDBConfig,
+* variable.TiDBEnableSlowLog,
+* variable.TiDBEnableTelemetry,
+* variable.TiDBExpensiveQueryTimeThreshold,
+* variable.TiDBForcePriority,
+* variable.TiDBGeneralLog,
+* variable.TiDBMetricSchemaRangeDuration,
+* variable.TiDBMetricSchemaStep,
+* variable.TiDBOptWriteRowID,
+* variable.TiDBPProfSQLCPU,
+* variable.TiDBRecordPlanInSlowLog,
+* variable.TiDBRowFormatVersion,
+* variable.TiDBSlowQueryFile,
+* variable.TiDBSlowLogThreshold,
+* variable.TiDBEnableCollectExecutionInfo,
+* variable.TiDBMemoryUsageAlarmRatio,
+
+The following system variables will be reset to defaults:
+
+* variable.Hostname
+
+### Status Variables
+
+The following status variables will be hidden:
+
+* tidb_gc_leader_desc
```

Comment:
@SunRunAway this section on Status Variables is new. It is based on the changes introduced in https://github.com/pingcap/tidb/pull/22286

Reply:
Will SRE be able to see this status?

Reply:
We will have to chose a privilege that gives them the ability to see it. Currently I have not listed one, but it is close to `RESTRICTED_SYSTEM_VARIABLES_ADMIN`. I can either include it in scope for this, or add a `RESTRICTED_STATUS_VARIABLES_ADMIN`.

Reply:
I prefer `RESTRICTED_STATUS_VARIABLES_ADMIN `.

Reply:
I've added it here https://github.com/pingcap/tidb/pull/23224/commits/5c92b62ad75e2762b6a9a21ea9647e8ee30de0d5 (it is listed in the other proposal)